packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -509,24 +514,21 @@ class TextSelectionOverlay {
     if (selectionControls == null)
       handle = Container();
     else {
-      handle = Visibility(
```

Comment:
There are two ways to control the handle visibilities, first from TextSelectionOverlay.handlesVisible and second from renderEditable.selectionStartHandleInViewport. This pr creates a _effectiveStartHandleVisibility to merge both api and feed it directly into  _SelectionHandleOverlay.

The difference is the _SelectionHandleOverlay use a fadeintransition instead of visibility widget. There might be a really minor visual difference.